src/Client/ClientBase.cpp

```diff
@@ -329,15 +330,24 @@ ASTPtr ClientBase::parseQuery(const char *& pos, const char * end, bool allow_mu
 
     const Dialect & dialect = settings.dialect;
 
-    if (dialect == Dialect::kusto)
-        parser = std::make_unique<ParserKQLStatement>(end, global_context->getSettings().allow_settings_after_format_in_insert);
-    else
-        parser = std::make_unique<ParserQuery>(end, global_context->getSettings().allow_settings_after_format_in_insert);
+    const auto *begin = pos;
 
     if (is_interactive || ignore_error)
     {
         String message;
-        res = tryParseQuery(*parser, pos, end, message, true, "", allow_multi_statements, max_length, settings.max_parser_depth);
+        if (dialect == Dialect::kusto)
+             res = tryParseQuery(kql_parser, pos, end, message, true, "", allow_multi_statements, max_length, settings.max_parser_depth);
+        else if (dialect == Dialect::kusto_auto)
+        {
+            res = tryParseQuery(parser, pos, end, message, true, "", allow_multi_statements, max_length, settings.max_parser_depth);
+            if (!res)
+            {
+                pos = begin;
+                res = tryParseQuery(kql_parser, pos, end, message, true, "", allow_multi_statements, max_length, settings.max_parser_depth);
+            }
+        }
```

Comment:
I'm not sure we want such complications, and after all it's possible we will have other dialects in the future...
@alexey-milovidov do you think we want to have autodetection of dialect? and if we want it I think it's better to generalize it to just `auto` to incorporate future dialects

Reply:
The `auto` option shouldn't be added in this PR, as it can be implemented (or not implemented at all) separately, later.

Reply:
@kashwy could you please remove this functionality for now - we will return to it some later

Reply:
@yakov-olkhovskiy , sure, will remove